eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/RatifySpec.hs

```diff
@@ -624,6 +627,117 @@ votingSpec =
           -- The same vote should now successfully ratify the proposal
           passEpoch
           getLastEnactedCommittee `shouldReturn` SJust (GovPurposeId addCCGaid)
+        describe "Proposal deposits contribute to active voting stake" $ do
+          it "Directly" $ do
+            -- Only modify the applicable thresholds
+            modifyPParams $ \pp ->
+              pp
+                & ppDRepVotingThresholdsL
+                  .~ def
+                    { dvtCommitteeNormal = 51 %! 100
+                    , dvtCommitteeNoConfidence = 51 %! 100
+                    }
+                & ppCommitteeMaxTermLengthL .~ EpochInterval 20
+                & ppGovActionDepositL .~ Coin 600_000
+            -- Setup DRep delegation without stake #1
+            (drepKH1, stakingKH1) <- setupDRepWithoutStake
+            -- Setup DRep delegation #2
+            (_drepKH2, _stakingKH2, _paymentKP2) <- setupSingleDRep 1_000_000
+            -- Make a note of the reward account for the delegator to DRep #1
+            dRepRewardAccount <- getRewardAccountFor $ KeyHashObj stakingKH1
+            -- Submit the first committee proposal, the one we will test active voting stake against.
+            -- The proposal deposit comes from the root UTxO
+            cc <- KeyHashObj <$> freshKeyHash
+            let addCCAction = UpdateCommittee SNothing mempty (Map.singleton cc 10) (75 %! 100)
+            addCCGaid <-
+              submitProposal $
+                ProposalProcedure
+                  { pProcDeposit = Coin 600_000
+                  , pProcReturnAddr = dRepRewardAccount
+                  , pProcGovAction = addCCAction
+                  , pProcAnchor = def
+                  }
+            -- Submit the vote from DRep #1
+            submitVote_ VoteYes (DRepVoter $ KeyHashObj drepKH1) addCCGaid
+            passNEpochs 2
+            -- The vote should not result in a ratification
+            isDRepAccepted addCCGaid `shouldReturn` False
+            getLastEnactedCommittee `shouldReturn` SNothing
```

Comment:
This is just a question for my education, that I was meaning to ask, because I came across it when doing some refactorings on the previous RP. 
I don't understand the meaning of these `getLastEnacted`* functions .
I can see they are looking in :
`nesEsL . esLStateL . lsUTxOStateL . utxosGovStateL . proposalsGovStateL . pRootsL`, and then `grCommitteeL . prRootL` for committee or `grConstitutionL . prRootL` for constitution, etc. 
So this would be the thing that *will* be enacted after one epoch?  What is the difference between this and looking in 
` nesEsL . esLStateL . lsUTxOStateL . utxosGovStateL . committeeGovStateL`  ?  Is the former a sort of preview of the latter?

Reply:
Great questions!

> So this would be the thing that will be enacted after one epoch?

No. These are the last enacted governance action ids, as the name suggests. Which means, they are each returning the last governance action id that was enacted in the last epoch boundary.

>  looking in `nesEsL . esLStateL . lsUTxOStateL . utxosGovStateL . committeeGovStateL`?

This will give you the current state of the committee, as a whole, but not tell you anything about what was the last governance action that was enacted at the previous epoch boundary.

Reply:
> I guess those two would coincide

The functions `getLastEnacted...` return a `GovActionId`, whereas the `Committee` is just the committee. That is also the difference. So they are not the same thing.